### PR TITLE
feature(sast): skip checks

### PR DESCRIPTION
**Pull request by achiar99**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

Implement skip checks for SAST

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/sast/checks/base_check.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from checkov.common.models.enums import CheckCategories
+
+
+class BaseSastCheck():
+    def __init__(self, name: str, id: str, severity = None) -> None:
+        self.name = name
+        self.id = id
+        self.categories = [CheckCategories.SAST]
+        # TODO
```
Comment (bo156) [inline checkov/sast/checks/base_check.py:9]: is this todo still needed? if so maybe expand what should be done?

File: checkov/sast/checks/base_registry.py [hunk 2]
```diff
@@ -46,8 +52,21 @@ def _load_checks_from_dir(self, directory: str, sast_languages: Set[SastLanguage
                         logging.warning(f'cant parse rule file {file}')
                         continue
                     for rule in rules:
+                        check = Registry._get_check_from_rule(rule)
+                        if not check:
+                            break
+                        if not self.runner_filter.should_run_check(check):
```
Comment (YaaraVerner) [inline checkov/sast/checks/base_registry.py:58]: the `runner_filter` value can be None (you set it in the constructor) so you will have an exception in this case

File: tests/sast/test_runner.py [hunk 3]
```diff
@@ -215,3 +215,15 @@ def test_get_generic_ast():
...
+
+
+def test_sast_skip_checks():
+    runner = Runner()
+    cur_dir = pathlib.Path(__file__).parent.resolve()
+    source = os.path.join(cur_dir, 'source_code')
+    external_dir_checks = os.path.join(cur_dir, 'external_checks')
+    report = runner.run(source, runner_filter=RunnerFilter(framework=['sast'], skip_checks=['CKV_SAST_1']), external_checks_dir=[external_dir_checks])
+    assert report.check_type == CheckType.SAST
+    assert len(report.failed_checks) == 2
```
Comment (bo156) [inline tests/sast/test_runner.py:227]: where do you check the skipped checks? 😕

**Review by bo156 — CHANGES_REQUESTED** [review]

**Review by YaaraVerner — APPROVED** [review]
had one comment 😊

File: checkov/sast/checks/base_check.py (hunk 1 above)
Comment (achiar99) [reply checkov/sast/checks/base_check.py:9]: We need to fill it in the future when we will have the final SAST checks format

File: tests/sast/test_runner.py (hunk 3 above)
Comment (achiar99) [reply tests/sast/test_runner.py]: We dont. the skipped checks is basically the unknown checks results (this is the logic in other runners..)

**Review by bo156 — APPROVED** [review]

File: checkov/sast/checks/base_check.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+from checkov.common.models.enums import CheckCategories
+
+
+class BaseSastCheck():
```
Comment (tronxd) [inline checkov/sast/checks/base_check.py:4]:
```suggestion
class BaseSastCheck(object):
```
or 
```suggestion
class BaseSastCheck:
```

File: checkov/sast/checks/base_registry.py [hunk 5]
```diff
@@ -30,6 +36,8 @@ def load_external_rules(self, dir: str, sast_languages: Optional[Set[SastLanguag
             self._load_checks_from_dir(dir, sast_languages)
 
     def _load_checks_from_dir(self, directory: str, sast_languages: Set[SastLanguages]) -> None:
+        if not self.runner_filter:
+            return
```
Comment (tronxd) [inline checkov/sast/checks/base_registry.py:40]: Why does a missing `runner_filter` means that we won't load any check?

**Pull request merged**